assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -601,15 +591,72 @@ const TemplateLibraryManager = function() {
 
 		data.content = JSON.stringify( data.content );
 
-		var ajaxParams = { data };
+		const ajaxParams = {
+			data,
+			success( successData ) {
+				$e.route( 'library/templates/my-templates', {
+					onBefore: () => {
+						self.triggerQuotaUpdate();
+						if ( templatesCollection ) {
+							const itemExist = templatesCollection.findWhere( {
+								template_id: successData.template_id,
+							} );
+
+							if ( ! itemExist ) {
+								templatesCollection.add( successData );
+							}
+						}
 
-		if ( templateType.ajaxParams ) {
-			_.extend( ajaxParams, templateType.ajaxParams );
-		}
+						self.sendOnSavedTemplateSuccessEvent( data );
+					},
+				} );
+			},
+			error( errorData ) {
+				self.showErrorDialog( errorData );
+				self.clearToastConfig();
+				self.sendOnSavedTemplateFailedEvent( data );
+			},
+		};
 
 		elementorCommon.ajax.addRequest( this.getSaveAjaxAction( data.save_context ), ajaxParams );
 	};
 
+	this.sendOnSavedTemplateSuccessEvent = ( formData ) => {
+		if ( SAVE_CONTEXTS.SAVE === formData.save_context ) {
+			return this.eventManager.sendNewSaveTemplateClickedEvent( {
+				library_type: formData.source,
+				template_type: formData.type,
+			} );
+		} else if ( [ SAVE_CONTEXTS.COPY, SAVE_CONTEXTS.MOVE ].includes( formData.save_context ) ) {
+			return this.eventManager.sendTemplateTransferEvent( {
+				transfer_method: formData.save_context,
+				template_type: formData.type,
+				template_origin: formData.from_source,
+				template_destination: formData.source,
+			} );
+		} else if ( [ SAVE_CONTEXTS.BULK_MOVE, SAVE_CONTEXTS.BULK_COPY ].includes( data.save_context ) ) {
+			self.eventManager.sendBulkActionsSuccessEvent( {
+				bulk_action: SAVE_CONTEXTS.BULK_MOVE === data.save_context ? 'move' : 'copy',
+				library_type: data.source,
+				bulk_count: data.from_template_id.length,
+				template_origin: formData.from_source,
+				template_destination: formData.source,
+			} );
+		}
+	};
+
+	this.sendOnSavedTemplateFailedEvent = ( formData ) => {
+		if ( [ SAVE_CONTEXTS.BULK_MOVE, SAVE_CONTEXTS.BULK_COPY ].includes( data.save_context ) ) {
```

Comment:
```suggestion
		if ( [ SAVE_CONTEXTS.BULK_MOVE, SAVE_CONTEXTS.BULK_COPY ].includes( formData.save_context ) ) {
```
Where does `data` come from? Check below as well.